File: API/Entities/Product.cs
using System;

namespace API.Entities
{
    public class Product
    {
        public int ProductId { get; set; }
        public string Name { get; set; }
        public string Category { get; set; }
        public string Description { get; set; }
        public string Type { get; set; }
        public long Price { get; set; }
        public string ImagePath { get; set; }
    }
}

File: shopping-cart/src/app/models/product.ts
export interface Product {
    productId: number;
    name: string;
    category: string;
    description: string;
    type: string;
    price: number;
    imagePath: string;
}

export interface ProductParams {
    orderBy: string;
    searchTerm?: string;
    types: string[];
    categories: string[];
    pageNumber: number;
    pageSize: number;
}